tests/test_keep-alive.py

```diff
@@ -0,0 +1,69 @@
+import gc
+
+
+def test_keep_alive_argument(capture):
+    from pybind11_tests import Parent, Child
+
+    with capture:
+        p = Parent()
+    assert capture == "Allocating parent."
+    with capture:
+        p.addChild(Child())
+        gc.collect()
+    assert capture == """
+        Allocating child.
+        Releasing child.
+    """
+    with capture:
+        del p
+        gc.collect()
+    assert capture == "Releasing parent."
+
+    with capture:
+        p = Parent()
+    assert capture == "Allocating parent."
+    with capture:
```

Comment:
This was my original implementation, however I ended up scraping it for a few (minor) reasons:

* To me it looked kind of weird with multiline strings (see example below).
* Using `assert_capture` for single-line and regular `capture` for multiline seemed inconsistent.
* There are other kinds of asserts (e.g. `assert "something" in capture`) which are more natural with the 'assert after' approach.

These aren't huge concerns, but there should actually be a minimal number of captures after the changes, so I don't think the line saving would be worth the inconsistency. I may be able to get rid of a few more of the captures -- I'll do another pass and see where `std::cout` can be replaced.

```python
with assert_capture("""
    Allocating child.
    Releasing child.
"""):
    p.addChild(Child())
    gc.collect()
 ```